src/utils.js

```diff
@@ -39,12 +41,20 @@ export function isWidget( element ) {
  * * sets `contenteditable` attribute to `true`,
  * * adds custom `getFillerOffset` method returning `null`,
  * * adds `ck-widget` CSS class,
- * * adds custom property allowing to recognize widget elements by using {@link ~isWidget}.
+ * * adds custom property allowing to recognize widget elements by using {@link ~isWidget},
+ * * implements `setVirtualSelection` and `removeVirtualSelection` custom properties to handle virtual selection
+ * on widgets.
  *
  * @param {module:engine/view/element~Element} element
  * @param {Object} [options={}]
  * @param {String|Function} [options.label] Element's label provided to {@link ~setLabel} function. It can be passed as
  * a plain string or a function returning a string.
+ * @param {Function} [setVirtualSelection] Function that will apply virtual selection on widget's element. When not
+ * provided, default behaviour will be used: adding CSS class from
+ * {@link module:engine/conversion/buildmodelconverter~VirtualSelectionDescriptor}.
+ * @param {Function} [removeVirtualSelection] Function that will remove virtual selection on widget's element. When not
+ * provided, default behaviour will be used: removing CSS class from
+ * {@link module:engine/conversion/buildmodelconverter~VirtualSelectionDescriptor}.
```

Comment:
I don't like such options. It changes `toWidget` to the uber-config with multiple functions and options. I would keep it simple. If one really need a custom handling of a widget, he does not need to use `toWidget`, can use all of it manually.